haskala/modules/custom/haskala_book/haskala_book.module

```diff
@@ -25,16 +26,91 @@ function haskala_book_ctools_plugin_directory($module, $plugin) {
  *
  *  @return production details nid.
  */
-function haskala_book_get_production($book_nid, $person_nid) {
+function haskala_book_get_production($book_nid, $person_nid = NULL) {
   $query = new EntityFieldQuery();
   $query->entityCondition('entity_type', 'node')
     ->entityCondition('bundle', 'production')
     ->propertyCondition('status', NODE_PUBLISHED)
-    ->fieldCondition('field_producer', 'target_id', $person_nid)
     ->fieldCondition('field_book', 'target_id', $book_nid);
+  if ($person_nid) {
+    $query->fieldCondition('field_producer', 'target_id', $person_nid);
+  }
   $result = $query->execute();
   if (empty($result['node'])) {
     return;
   }
-  return reset($result['node'])->nid;
+  if ($person_nid) {
+    return reset($result['node'])->nid;
+  }
+  return $result['node'];
+}
+
+/**
+ *  Tabs with data from another bundles.
+ *
+ *  @param $bundle_name
+ *   The bundle name.
+ *  @param $node
+ *   The node object.
+ * @param $tab_name
+ *   The tab name for group info.
+ *
+ *  @return production details nid.
+ */
+function haskala_book_get_bundle_tab($bundle_name, $node, $tab_name) {
```

Comment:
Let's drop this function. Please move these calls to where this function is called.